File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringBuffer sb = new StringBuffer();
        Stack<Integer> stack = new Stack<>();
        int cnt = 1;
        for (int i = 0; i < n; i++) {
            int k = Integer.parseInt(br.readLine());
            while (cnt <= k) {
                stack.push(cnt);
                sb.append("+\n");
                cnt++;
            }
            if (stack.peek() != k) {
                System.out.println("NO");
                System.exit(0);
            }
            stack.pop();
            sb.append("-\n");
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
from sys import stdin
input = stdin.readline

n = int(input())
stack, answer = [], []
cnt = 1
for _ in range(n):
    k = int(input())
    while cnt <= k:
        stack.append(cnt)
        answer.append("+")
        cnt += 1
    if stack[-1] != k:
        print("NO")
        exit()
    stack.pop()
    answer.append("-")
print('\n'.join(answer))
    
